website/static/js/components/quickFiles.js

```diff
@@ -0,0 +1,151 @@
+var $osf = require('js/osfHelpers');
+var iconmap = require('js/iconmap');
+var lodashFind = require('lodash.find');
+var mHelpers = require('js/mithrilHelpers');
+var Raven = require('raven-js');
+
+var withPagination = require('js/components/pagination').withPagination;
+
+var QUICKFILES_PAGE_SIZE = 10;
+
+
+var _buildUrl = function(page, user) {
+
+    var query = {
+        'page[size]': QUICKFILES_PAGE_SIZE,
+        'page': page || 1,
+        'version': '2.2',
+    };
+
+    return $osf.apiV2Url('users/' + user +  '/quickfiles/', { query: query});
+};
+
+
+var _getNextItems = function(ctrl, url, updatePagination) {
+    if(ctrl.requestPending()) {
+        return;
+    }
+
+    ctrl.quickFiles([]);
+    ctrl.requestPending(true);
+
+    var promise = m.request({
+        method : 'GET',
+        url : url,
+        background : true,
+        config: mHelpers.apiV2Config({withCredentials: window.contextVars.isOnRootDomain})
+    });
+
+    promise.then(
+        function(result) {
+            ctrl.requestPending(false);
+            ctrl.quickFiles(result.data);
+            updatePagination(result, url);
+            m.redraw();
+            return promise;
+        }, function(xhr, textStatus, error) {
+            ctrl.failed = true;
+            ctrl.requestPending(false);
+            m.redraw();
+            Raven.captureMessage('Error retrieving quickfiles', {
+                extra: {
+                    url: url,
+                    textStatus: textStatus,
+                    error: error
+                }
+            });
+        }
+    );
+};
+
+
+var QuickFile = {
+
+    controller: function(options) {
+        var self = this;
+        self.file = options.file;
+        self.icon =  iconmap.file;
+    },
+
+    view: function(ctrl)  {
+        var viewBase = window.location.origin + '/quickfiles';
+        var viewUrl = ctrl.file.attributes.guid ? viewBase + '/' + ctrl.file.attributes.guid : viewBase + ctrl.file.attributes.path;
+        return m('div', [
+            m('li.project list-group-item list-group-item-node cite-container', [
+                m('h4.list-group-item-heading', [
+                    m('span.component-overflow.f-w-lg', {style: 'line-height: 1.5; width: 100%'}, [
+                        m('span.project-statuses-lg', {style: 'width: 50%; float:left'}, [
+                            m('span', {class: ctrl.icon, style: 'padding-right: 5px;'}, ''),
+                            m('a', {'href': viewUrl,
+                                onclick : function () {
+                                    $osf.trackClick('QuickFiles', 'view', 'view-quickfile-from-profile-page');
+                                }
+                            }, ctrl.file.attributes.name),
+                        ]),
+                        m('div', {style: {width: '50%', float: 'right', fontSize: 'small', marginTop: '2%'}}, 'downloads: ' + ctrl.file.attributes.extra.downloads)
```

Comment:
Drive-by comment: Might want to use an object consistently above.

Reply:
normalized!